src/dune_rules/modules.ml

```diff
@@ -741,39 +746,93 @@ and impl =
   ; vlib : t
   }
 
+module Sourced_module = struct
+  type t =
+    | Normal of Module.t
+    | Imported_from_vlib of Module.t
+    | Impl_of_virtual_module of Module.t Ml_kind.Dict.t
+end
+
+let rec obj_map :
```

Comment:
This function and `obj_map_build` have not changed, just moved up due to scoping issues.